spec/asyncapi.md

```diff
@@ -815,6 +817,8 @@ Field Name | Type | Description
 <a name="operationObjectExternalDocs"></a>externalDocs | [External Documentation Object](#externalDocumentationObject) \| [Reference Object](#referenceObject) | Additional external documentation for this operation.
 <a name="operationObjectBindings"></a>bindings | [Operation Bindings Object](#operationBindingsObject) \| [Reference Object](#referenceObject) | A map where the keys describe the name of the protocol and the values describe protocol-specific definitions for the operation.
 <a name="operationObjectTraits"></a>traits | [[Operation Trait Object](#operationTraitObject) &#124; [Reference Object](#referenceObject) ] | A list of traits to apply to the operation object. Traits MUST be merged into the operation object using the [JSON Merge Patch](https://tools.ietf.org/html/rfc7386) algorithm in the same order they are defined here.
+<a name="operationObjectMessages"></a>messages | [[Reference Object](#referenceObject)] | A list of `$ref` pointers pointing to the supported [Message Objects](#messageObject) that can be processed by this operation. It MUST contain a subset of the messages defined in the [channel referenced in this operation](#operationObjectChannel). **Every message processed by this operation MUST be valid against one, and only one, of the [message objects](#messageObject) referenced in this list.** Please note the `messages` property value MUST be a list of [Reference Objects](#referenceObject) and, therefore, MUST NOT contain [Message Objects](#messageObject). However, it is RECOMMENDED that parsers (or other software) dereference this property for a better development experience.
+<a name="operationObjectReply"></a>reply | [Reply Object](#operationReplyObject) | The definition of the reply in a request-reply operation.
```

Comment:
We can define replies object in component we should also allow to use reference here:

```suggestion
<a name="operationObjectReply"></a>reply | [Operation Reply Object](#operationReplyObject) &#124; [Reference Object](#referenceObject)  | The definition of the reply in a request-reply operation.
```

Reply:
Can the parser handle this? Any why it is different for CorrelationId ?

Reply:
@magicmatatjahu can you please provide a sample of how a reference would look like for:
https://github.com/GreenRover/spec/blob/request-reply/examples/adeo/requestReply.yaml#L133

Reply:
@GreenRover 

Yes, parser can handle that, don't worry, but my problem is that, we can define generic/reusable reply object in `components.replies.[X]` but in the `operations.[x].reply` we cannot use `$ref` keyword, so we should decide:
- we can define replies in components and we also add ability to make references in `operations.[x].reply`
- we remove `replies` from components and people won't be able to use `$ref` in `operations.[x].reply`

I opt for first option, to add `$ref` ability in operation's reply.

About your question about `correlationId`. What do you have in mind? You can defined `correlationId` in message as inline object and ref:

![image](https://user-images.githubusercontent.com/20404945/216284114-e0df331f-f6f7-40c1-812f-8414c7c5eef0.png)

And here is a example:

```yaml
operations:
  requestCosting:
    action: receive
    channel: 
      $ref: '#/channels/costingRequest'
    reply:
      $ref: '#/components/replies/requestCostingReply'
          
 components:
   replies:
     requestCostingReply:
        channel: 
          $ref: '#/channels/costingResponse'
        address:
          locaton: '$message.header#/REPLY_TOPIC'
```

Let me know about concerns!